### PR TITLE
add useRef

### PR DESCRIPTION
**Pull request by jssckbl**
## Description

We merged the branch `jw-ac-fix-drag-map` with `main`, when we should have first attempted to merge `main` into the branch `jw-ac-fix-drag-map`. 

This would have potentially helped work through any expected conflicts, without adversely affecting production in the process.

This pull request is bringing the code back to it's intended state for `jw-ac-fix-drag-map`. 
Once this PR approved, we will merge `main` INTO `jw-ac-fix-drag-map`, and then solve any resulting conflicts. 
Once the conflicts are solved, we will be able to merge `jw-ac-fix-drag-map` into `main`

## Related Issue

Reference [TCL30-18](https://github.com/the-collab-lab/tcl-30-whats-near-me/pull/18/files).

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |
|   ✓  | :repeat: Same code, New PR     |

### Before

See Reference [TCL30-18](https://github.com/the-collab-lab/tcl-30-whats-near-me/pull/18).

### After

See Reference [TCL30-18](https://github.com/the-collab-lab/tcl-30-whats-near-me/pull/18).

## Testing Steps / QA Criteria

See Reference [TCL30-18](https://github.com/the-collab-lab/tcl-30-whats-near-me/pull/18).

### CODE REVIEW

File: src/components/ShareMyLocationButton.js [hunk 1]
```diff
@@ -19,41 +19,30 @@ const ShareMyLocationButton = () => {
...
-      const trackingId = navigator.geolocation.watchPosition(
-        handleSuccess,
-        handleError,
-      );
-
-      setTrackingId(trackingId);
-    }
-  };
-
-  const handleStopSharing = () => {
```
Comment (laurenmbeatty) [inline src/components/ShareMyLocationButton.js]: We're still going to need this `handleStopSharing` function.

File: src/components/ShareMyLocationButton.js [hunk 2]
```diff
@@ -19,41 +19,30 @@ const ShareMyLocationButton = () => {
...
+
+        setTrackingId(trackingId);
+      }
     }
-    setUserLocationShared(false);
-    setNewCenterMap(defaultCenterMap);
   };
 
   return (
-    <button
```
Comment (laurenmbeatty) [inline src/components/ShareMyLocationButton.js]: We'll also need to preserve these changes, including the `buttonText`. Once the user shares their location, the button text should change to "Stop Sharing Location", and the `onClick` function to change to `handleStopSharing`.

**Review by laurenmbeatty — CHANGES_REQUESTED** [review]

File: src/components/ShareMyLocationButton.js [hunk 3]
```diff
@@ -27,6 +28,10 @@ const ShareMyLocationButton = ({ disabled = false }) => {
   };
   const handleError = () => setUserCenterMap(defaultCenterMap);
 
+  const buttonText = userLocationShared
```
Comment (alxmcr) [inline src/components/ShareMyLocationButton.js]: Lauren : We added `buttonText` 😄  .

File: src/components/ShareMyLocationButton.js [hunk 4]
```diff
@@ -35,30 +40,23 @@ const ShareMyLocationButton = ({ disabled = false }) => {
...
-        onClick={handleShareLiveLocation}
-        disabled={sharingLocation.current}
-      >
-        Share My Location
-      </button>
-      {sharingLocation.current ? (
-        <button onClick={handleStopSharing}>Stop Sharing Location</button>
-      ) : null}
-    </>
+    <button
```
Comment (alxmcr) [inline src/components/ShareMyLocationButton.js]: Lauren: We added their button 😄 .

**Comment by alxmcr** [comment]
I'm so sorry for the inconvenience.

**Extra Comments:**
- We brought Katia & Megan's work to our branch (the last version) and added ours changes.
- We fixed the bug when we drag the map and change the map center location.
- I added two specific changes (after our previous PR): first,  remove the props for this component, because Katia & Megan thought a better idea, and second, there was an issue when you share, stop and try sharing your location again (it wasn't possible), inside `handleStopSharing` function we update `sharingLocation`'s value. It's fixed. (cc @jssckbl)

**Review by laurenmbeatty — APPROVED** [review]
yay!!!!!!! This is working beautifully now! Just tried it out on my phone. Let's get this merged! :shipit:

**Pull request merged**